exchange/utils.go

```diff
@@ -600,38 +600,23 @@ func isBidderInExtAlternateBidderCodes(adapter, currentMultiBidBidder string, ad
 
 // extractBuyerUIDs parses the values from user.ext.prebid.buyeruids, and then deletes those values from the ext.
 // This prevents a Bidder from using these values to figure out who else is involved in the Auction.
-func extractBuyerUIDs(user *openrtb2.User) (map[string]string, error) {
-	if user == nil {
-		return nil, nil
-	}
-	if len(user.Ext) == 0 {
+func extractBuyerUIDs(req *openrtb_ext.RequestWrapper) (map[string]string, error) {
+	if req.User == nil {
 		return nil, nil
 	}
-
-	var userExt openrtb_ext.ExtUser
-	if err := jsonutil.Unmarshal(user.Ext, &userExt); err != nil {
+	userExt, err := req.GetUserExt()
+	if err != nil {
 		return nil, err
 	}
-	if userExt.Prebid == nil {
+	prebid := userExt.GetPrebid()
+	if prebid == nil {
 		return nil, nil
 	}
+	userExt.SetPrebid(nil)
```

Comment:
This has some potential to bite us if `user.ext.prebid` is expanded to include other stuff in the future. But we should catch that pretty quickly if that does happen.

Reply:
Agreed I considered that point. Do you think it would be better to just set `BuyerUIDs` to nil? I'm not sure if that would ultimately end up in `Prebid` being removed when rebuilding the request which is our current behavior.